src/sql/workbench/contrib/objectExplorer/test/browser/connectionTreeActions.test.ts

```diff
@@ -547,6 +563,200 @@ suite('SQL Connection Tree Action tests', () => {
+		connectionManagementService.setup(x => x.getConnectionGroups()).returns(() => [conProfGroup]);
+		connectionManagementService.setup(x => x.getActiveConnections()).returns(() => [connection]);
+		connectionManagementService.setup(x => x.addSavedPassword(TypeMoq.It.isAny())).returns(() => new Promise<ConnectionProfile>((resolve) => {
+			resolve(connection);
+		}));
+		connectionManagementService.setup(x => x.isConnected(undefined, TypeMoq.It.isAny())).returns(() => isConnectedReturnValue);
+
+		let objectExplorerSession = {
+			success: true,
+			sessionId: '1234',
+			rootNode: {
+				nodePath: 'testServerName\tables',
+				nodeType: NodeType.Folder,
+				label: 'Tables',
+				isLeaf: false,
+				metadata: null,
+				nodeSubType: '',
+				nodeStatus: '',
+				errorMessage: ''
+			},
+			errorMessage: ''
+		};
+
+		let tablesNode = new TreeNode(NodeType.Folder, 'Tables', false, 'testServerName\Db1\tables', '', '', null, null, undefined, undefined);
+		tablesNode.connection = connection;
+		tablesNode.session = objectExplorerSession;
+		let table1Node = new TreeNode(NodeType.Table, 'dbo.Table1', false, 'testServerName\tables\dbo.Table1', '', '', tablesNode, null, undefined, undefined);
+		let table2Node = new TreeNode(NodeType.Table, 'dbo.Table1', false, 'testServerName\tables\dbo.Table1', '', '', tablesNode, null, undefined, undefined);
+		tablesNode.children = [table1Node, table2Node];
+		let objectExplorerService = TypeMoq.Mock.ofType(ObjectExplorerService, TypeMoq.MockBehavior.Loose, connectionManagementService.object);
+		objectExplorerService.callBase = true;
+		objectExplorerService.setup(x => x.getObjectExplorerNode(TypeMoq.It.isAny())).returns(() => tablesNode);
+		objectExplorerService.setup(x => x.refreshTreeNode(TypeMoq.It.isAny(), TypeMoq.It.isAny())).returns(() => Promise.resolve([table1Node, table2Node]));
+		const treeMock = TypeMoq.Mock.ofType(AsyncServerTree, TypeMoq.MockBehavior.Strict,
+			'ConnectionTreeActionsTest', // user
+			$('div'), // container
+			{}, // delegate
+			[], // renderers
+			{}, // data source
+			{}, // options
+			new MockContextKeyService(), // IContextKeyService
+			new TestListService(), // IListService,
+			undefined, // IThemeService,
+			new TestConfigurationService(), // IConfigurationService,
+			undefined, // IKeybindingService,
+			new TestAccessibilityService()); // IAccessibilityService
+		treeMock.callBase = true;
+		treeMock.setup(x => x.expand(TypeMoq.It.isAny())).returns(() => Promise.resolve(null));
+		treeMock.setup(x => x.collapse(TypeMoq.It.isAny())).returns(() => true);
+		treeMock.setup(x => x.updateChildren(TypeMoq.It.isAny())).returns(() => Promise.resolve());
+		let refreshAction: RefreshAction = new RefreshAction(RefreshAction.ID,
+			RefreshAction.LABEL,
+			treeMock.instance,
+			connection,
+			connectionManagementService.object,
+			objectExplorerService.object,
+			undefined,
+			logService);
+
+		return refreshAction.run().then((value) => {
```

Comment:
If you're writing these new tests, would it make sense to make them use async-await?

Reply:
Yeah I'll try to do that. Definitely plan on doing a lot more of that kind of clean up when I remove the old tree stuff too so may just wait for that to happen though.